src/bindings/ScriptDev2/scripts/kalimdor/caverns_of_time/hyjal/hyjalAI.cpp View
@@ -284,7 +284,7 @@ void hyjalAI::SpawnCreatureForWave(uint32 uiMobEntry)
			m_creature->GetRandomPoint(pSpawn->m_fX, pSpawn->m_fY, pSpawn->m_fZ, 40.0f, fX, fY, fZ);
			m_creature->SummonCreature(uiMobEntry, fX, fY, fZ + 30.0f, 10.0f, TEMPSUMMON_TIMED_DESPAWN, 15000);
		}else
-			m_creature->SummonCreature(uiMobEntry, pSpawn->m_fX, pSpawn->m_fY, pSpawn->m_fZ, 5.0f, TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 120000);
+			m_creature->SummonCreature(uiMobEntry, pSpawn->m_fX, pSpawn->m_fY, pSpawn->m_fZ, 5.0f, TEMPSUMMON_DEAD_DESPAWN, 120000);
	}
}

@@ -473,6 +473,8 @@ void hyjalAI::SummonedCreatureJustDied(Creature* pSummoned)
	m_pInstance->DoUpdateWorldState(WORLD_STATE_ENEMY, 0);

	m_creature->SetActiveObjectState(false);
+
+	lWaveMobGUIDList.clear();
}

void hyjalAI::SummonNextWave()
@@ -715,6 +717,8 @@ void hyjalAI::JustDied(Unit* pKiller)
			if (Creature* pTemp = m_pInstance->instance->GetCreature(*itr))
				pTemp->ForcedDespawn();
		}
+
+		lWaveMobGUIDList.clear();
	}
}

@@ -723,22 +727,8 @@ void hyjalAI::UpdateAI(const uint32 uiDiff)
	if (!m_bIsEventInProgress)
		return;

-	if (m_bIsSummoningWaves && m_pInstance)
+	if (m_bIsSummoningWaves)
	{
-		if (m_uiWaveMoveTimer < uiDiff)
-		{
-			// Skip the master timer, and start next wave in 5. Clear the list, it should not be any here now.
-			if (!m_pInstance->GetData(TYPE_TRASH_COUNT))
-			{
-				lWaveMobGUIDList.clear();
-				m_uiNextWaveTimer = std::min(m_uiNextWaveTimer, (uint32)5000);
-			}
-
-			m_uiWaveMoveTimer = 10000;
-		}
-		else
-			m_uiWaveMoveTimer -= uiDiff;
-
		if (m_uiNextWaveTimer < uiDiff)
			SummonNextWave();
		else